src/main/java/org/hyperskill/hstest/stage/SpringTest.java

```diff
@@ -172,12 +177,35 @@ public static void startSpring() throws Exception {
                                 .map(Class::getCanonicalName)
                                 .collect(Collectors.joining(", "))
                 );
+            } else if (!ReflectionUtils.hasMainMethod(suitableClasses.get(0)) && !isKotlin) {
+                throw new ErrorWithFeedback("The main method was not found in the class"
+                        + suitableClasses.get(0).getSimpleName());
             }
 
-            Class<?> springClass = suitableClasses.get(0);
-            Method mainMethod = ReflectionUtils.getMainMethod(springClass);
-            mainMethod.invoke(null, new Object[] {args});
-            springRunning = true;
+            if (!isKotlin) {
+                ReflectionUtils.getMainMethod(suitableClasses.get(0))
+                        .invoke(null, new Object[]{args});
+                springRunning = true;
+            } else {
+                List<Class<?>> allClassesFromPackage = ReflectionUtils.getAllClassesFromPackage("");
+                allClassesFromPackage.forEach(it -> {
+                    if (it.getCanonicalName().endsWith("Kt")) {
+                        List<Method> listOfMethods = Arrays.stream(it.getDeclaredMethods()).collect(Collectors.toList());
+                        listOfMethods.forEach(method -> {
+                            if (method.getName().equals("main")) {
```

Comment:
There's a better way to check for main method - `ReflectionUtils.hasMainMethod(it)` - why do you need to check every method in all classes?

Also, from this code, I suspect that it will invoke every `main` method in every class. I think it should invoke only one of the main methods?

Reply:
`List<Class<?>> allClassesFromPackage = ReflectionUtils.getAllClassesFromPackage("");
                allClassesFromPackage.forEach(it -> {
                    if (it.getCanonicalName().endsWith("Kt")
                            && ReflectionUtils.hasMainMethod(it)
                            && !springRunning) {
                        try {
                            ReflectionUtils.getMainMethod(it)
                                    .invoke(null, new Object[]{args});
                            springRunning = true;
                        } catch (IllegalAccessException | InvocationTargetException e) {
                            throw new ErrorWithFeedback("The main method was not found in the class" +
                                    it.getSimpleName());
                        }
                    }
                });`

I used the hasMainMethod (it) method and checking the springRunning flag, if the application starts, I will no longer check the classes.